### ISSUE TITLE
catt is slow on Raspberry Pi 1

### ISSUE THREAD

**Issue by 6b6561** [skorokithakis/catt#357]
This is not a bug report, it's more a question on what to expect. I don't expect miracles from an old Raspberry, but I would have expected it to be a bit faster, 12s to display the version of catt doesn't sound really right. Is it just this slow or should I look for problems?

Any ideas what I could check? Or should I just retire the old Pi and get a generation 4 to replace it.

```
kea@hall2:~$ time catt --version
catt v0.12.2, Zaniest Zapper.

real    0m12.773s
user    0m11.773s
sys     0m0.836s
kea@hall2:~$
```

catt really sticks out in top output.
```
  PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
  908 kea       20   0   24560  16216   7344 R  95.5   3.4   0:03.63 catt

```

```
kea@hall2:~$ free
              total        used        free      shared  buff/cache   available
Mem:         472840       77644      208676        9712      186520      291848
Swap:        102396           0      102396
kea@hall2:~$

```

**Comment by skorokithakis** [comment]
Hmm, no, this is definitely not normal. How long does it take to start the Python interpreter?

**Comment by 6b6561** [comment]
I know my way around linux, but Python is not really my bread and butter... but I'm trying... Hope that the following is somewhat what you are asking.

```
kea@hall2:~$ time python3.7 -c pass

real    0m0.552s
user    0m0.486s
sys     0m0.059s
kea@hall2:~$
```

**Comment by skorokithakis** [comment]
Yep, that's exactly it. That's extremely odd, can you test how long it takes to display the Catt version if you disconnect the network? I suspect it's waiting for something.

**Comment by 6b6561** [comment]
Always refreshing to stop the network on a headless device... wrote a small script, crossed my fingers and ran it blindly, result is the same 12.2 seconds real time.

**Comment by skorokithakis** [comment]
Yeah, that's really odd. I'll have a look and see if I spot something, thanks.

**Comment by 6b6561** [comment]
Don't put to much time on it, I start to suspect that it's just slow... trying to find some python app that I could install to benchmark the box.

**Comment by 6b6561** [comment]
I'm sorry for not doing my homework prior to starting this thread! My raspberry is just painfully slow, a "pip3 list" takes ~30s to complete. 

I have been using catt for quite a while from my NAS, it's the backend of a shell script for casting preset radio channels, mp3/flac library to my devices.... It runs 8h per day here in my home office and I have realized that I'm very dependent on it, therefore I went ahead to setup a backup instance on a old raspberry which otherwise performance quite well, obviously it wasn't up to the task.

A big thank you for catt, giving me full control of what's playing on my chromecasts!
